modules/launcher/src/main/java/org/mule/module/launcher/MuleContainer.java

```diff
@@ -135,6 +136,10 @@ protected void init(String[] args) throws IllegalArgumentException
         {
             setStartupPropertiesFile(propertiesFile);
         }
+
+        JdbcDriverManagerInitializer jdbcDriverManagerInitializer = new JdbcDriverManagerInitializer();
```

Comment:
DriverManager uses Service Provider for loading JDBC 4.0 drivers as part of DriverManager class initialization.

If an application didn't use the driver (didn't get a connection from DriverManager neither did a Class.forName("driverName")) when application classloader is disposed the DriverManager is called for the first time therefore it was loading all the drivers found by SPI before they were just about to be deregistered.

This "side-effect" ended up locking the OracleDriver jar due to OracleDriver leaves inputStreams open to get error mappings from properties file. But the GoodCitizenClassLoader was doing a workaround for closing any opened jar and resources, instead now calling URLClassLoader.close() (method added in Java 7) it does a better job and resources are released and the Oracle jar file is not locked and those input streams closed.

I found this after researched how DriverManager could be handled. I just validated and  confirmed that it is not needed to do the DriverManager initialization in order to fix this issue. 
I don't see that avoiding loading drivers when not used and are about to be deregistered could have any other issue now that the classloaders are disposed correctly.